OpenRA.Mods.Common/Widgets/SupportPowerTimerWidget.cs

```diff
@@ -19,47 +19,155 @@ namespace OpenRA.Mods.Common.Widgets
 {
 	public class SupportPowerTimerWidget : Widget
 	{
+		public readonly int XObserver = 10;
+		public readonly int YObserver = 10;
+		public enum TimerOrder { AlliesFirst, EnemiesFirst }
+		public readonly TimerOrder Order = TimerOrder.EnemiesFirst;
+		[Translate] public readonly string HeaderAllies = "Allies";
+		[Translate] public readonly string HeaderEnemies = "Enemies";
 		public readonly string Font = "Bold";
 		public readonly string Format = "{0}: {1}";
-		public readonly TimerOrder Order = TimerOrder.Descending;
 
-		readonly IEnumerable<SupportPowerInstance> powers;
-		Pair<string, Color>[] texts;
+		Dictionary<Player, SupportPowerManager> init;
+		struct TextBlock
+		{
+			public Pair<string, Color> LeftColumn;
+			public Pair<string, Color>[] RightColumn;
+		}
+
+		TextBlock t, ha, he;
+		List<TextBlock> texts;
+		float2 b, boundsPlayer, boundsObserver;
+		readonly World world;
+		int lastTick;
 
 		[ObjectCreator.UseCtor]
 		public SupportPowerTimerWidget(World world)
 		{
-			powers = world.ActorsWithTrait<SupportPowerManager>()
-				.Where(p => !p.Actor.IsDead && !p.Actor.Owner.NonCombatant)
-				.SelectMany(s => s.Trait.Powers.Values)
-				.Where(p => p.Instances.Any() && p.Info.DisplayTimer && !p.Disabled);
+			t = new TextBlock();
+
+			ha = new TextBlock();
+			ha.LeftColumn = new Pair<string, Color>(HeaderAllies, Color.White);
+
+			he = new TextBlock();
+			he.LeftColumn = new Pair<string, Color>(HeaderEnemies, Color.White);
+
+			boundsPlayer = new float2(Bounds.Location);
```

Comment:
This doesn't work. Bounds.Location has the correct values but boundsPlayer is zero. Why?